apps/modernization-ui/src/apps/page-builder/pages/BusinessRulesLibrary/Add/AddBusinessRule.tsx

```diff
@@ -0,0 +1,199 @@
+import { Button, ButtonGroup, Form, Grid, Icon } from '@trussworks/react-uswds';
+import { useEffect, useState } from 'react';
+import { FormProvider, useForm } from 'react-hook-form';
+import { useNavigate, useParams } from 'react-router-dom';
+import './AddBusinessRule.scss';
+import { PageRuleControllerService, ViewRuleResponse } from '../../../generated';
+import { useAlert } from 'alert';
+import BusinessRulesForm from '../BusinessRulesForm';
+import { PageBuilder } from '../../PageBuilder/PageBuilder';
+import { Breadcrumb } from 'breadcrumb';
+import { authorization } from 'authorization';
+
+export type FormValues = {
+    ruleFunction: string;
+    sourceIdentifier: string;
+    sourceText: string;
+    comparator?: string;
+    sourceValueIds?: string[];
+    sourceValueText?: string[];
+    sourceValue?: {
+        sourceValueId?: string[];
+        sourceValueText?: string[];
+    };
+    targetValueIdentifier?: string[];
+    targetType: string;
+    ruleDescription: string;
+    targetValueText?: string[];
+    anySourceValue?: boolean;
+};
+
+const AddBusinessRule = () => {
+    const navigate = useNavigate();
+    const form = useForm<FormValues>({
```

Comment:
Suggestion: Instead of creating a new type for the form,  you can bind the form directly to the object required by the API, which i think would be `useForm<CreateRuleRequest>` in this case. 


This would remove the need to do 
```typescript
 const request = {
            anySourceValue: data.anySourceValue,
            comparator: data.comparator,
            ruleDescription: data.ruleDescription,
            ruleFunction: data.ruleFunction,
            sourceIdentifier: data.sourceIdentifier,
            sourceText: data.sourceText,
            sourceValue: sourceValue,
            targetType: data.targetType,
            targetValueIdentifier: data.targetValueIdentifier,
            targetValueText: data.targetValueText
        };
```
within the submit method and instead you could directly pass the form data:

```typescript
 PageRuleControllerService.createBusinessRuleUsingPost({
                authorization: authorization(),
                id: Number(pageId),
                request: data
            }).then((resp) => {
                console.log('resp', resp);
                showAlert({ type: 'success', header: 'added', message: resp.message });
            });
```

Reply:
I did this, and generated the Api models again.  but we have a problem that I think the API response is missing, "anyRuleResponse"